pandas/core/frame.py

```diff
@@ -1370,6 +1370,8 @@ def to_numpy(
         result = self._mgr.as_array(
             transpose=self._AXIS_REVERSED, dtype=dtype, copy=copy, na_value=na_value
         )
+        if result.dtype is not dtype:
+            result = np.array(result, dtype=dtype, copy=False)
```

Comment:
you need to use pd.array here

Reply:
Don't we want to return a numpy array rather than a pandas array (method is to_numpy)?